airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -421,6 +421,26 @@ public List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(final J
     final List<CheckedConsumer<JdbcDatabase, Exception>> checkOperations = new ArrayList<>(
         super.getCheckOperations(config));
 
+    // if UI has schemas specified, check if the user has select access to any table
+    if (schemas != null && !schemas.isEmpty()) {
```

Comment:
I think, but I'm not sure, that if `schemas` is null or empty, we fall back to `public`. In that case, we should also be performing these checks you're adding.

Reply:
yes, i just learned from our ui that `public` will be the default is none is specified.